### PR TITLE
Fixed: Download issues

### PR DESCRIPTION
**Pull request by HardResetSolution**
- Merging under Windows was broken.
- Youtube-dl selects video parts with different resolutions which then cannot be merged.
- In German S22E08 and S22E10 audio and video tracks are mixed up and can also not be merged.

### CODE REVIEW

File: utils/downloader.js [hunk 1]
```diff
@@ -39,10 +41,75 @@ module.exports.downloadEpisode = async function downloadEpisode(season, episode,
...
+            reject()
+        })
+
+        command.on('close', (code) => {
+            var lines = data.split('\n');
+            var myRegexp = /(\d{3,4})x\d{3,4}/;
+
+            var allResolutions = [];
+
+            for(var i = 0;i < lines.length;i++){
```
Comment (flokol120) [inline utils/downloader.js]: please insert spaces after each ; (code style)

File: utils/downloader.js [hunk 2]
```diff
@@ -39,10 +41,75 @@ module.exports.downloadEpisode = async function downloadEpisode(season, episode,
...
+               var line = lines[i].match(myRegexp);
+
+               if(line != null)
+               {
```
Comment (flokol120) [inline utils/downloader.js]: please move the opening curly bracket to the end of the if statement separated by a space (code style)

File: utils/downloader.js [hunk 3]
```diff
@@ -39,10 +41,75 @@ module.exports.downloadEpisode = async function downloadEpisode(season, episode,
...
+
+            var resolutionTriplets = [];
+            for (var i = 0; i < allResolutions.length - 2; i++) {
+                if (allResolutions[i + 1] == allResolutions[i] && allResolutions[i + 2] == allResolutions[i]) {
+                    resolutionTriplets.push(allResolutions[i]);
+                }
+            }
+
+            if(resolutionTriplets.length == 0)
+            {
```
Comment (flokol120) [inline utils/downloader.js]: same as above (code style)

File: utils/downloader.js [hunk 4]
```diff
@@ -189,7 +192,92 @@ async function download(link, path, output, season, episode, resolutionWidth, pr
...
+            var lines = data.split('\n');
+            var myRegexp = /Track ID (\d): (video|audio)/;
+            var video;
+            var audio;
+
+            for(var i = 0;i < lines.length;i++){
+                var line = lines[i].match(myRegexp);
+
+                if(line == null)
+                {
```
Comment (flokol120) [inline utils/downloader.js]: please move the opening curly bracket to the end of the if statement separated by a space (code style)

File: utils/downloader.js [hunk 5]
```diff
@@ -189,7 +192,92 @@ async function download(link, path, output, season, episode, resolutionWidth, pr
...
+            reject()
+        })
+
+        command.on('close', (code) => {
+            var lines = data.split('\n');
+            var myRegexp = /Track ID (\d): (video|audio)/;
+            var video;
+            var audio;
+
+            for(var i = 0;i < lines.length;i++){
```
Comment (flokol120) [inline utils/downloader.js]: please insert spaces after each ; (code style)

**Comment by flokol120** [comment]
Thank you for your PR @HardResetSolution! Looks good to me except for some minor code styling differences. Please quickly fix those and I will go ahead and merge your promising changes!

File: utils/downloader.js (hunk 5 above)
Comment (HardResetSolution) [reply utils/downloader.js]: sorry for the sloppy code styling.

File: utils/downloader.js (hunk 4 above)
Comment (HardResetSolution) [reply utils/downloader.js]: done

File: utils/downloader.js (hunk 2 above)
Comment (HardResetSolution) [reply utils/downloader.js]: done

File: utils/downloader.js (hunk 1 above)
Comment (HardResetSolution) [reply utils/downloader.js]: done

File: utils/downloader.js (hunk 3 above)
Comment (HardResetSolution) [reply utils/downloader.js]: done

File: utils/downloader.js [hunk 6]
```diff
@@ -189,7 +192,92 @@ async function download(link, path, output, season, episode, resolutionWidth, pr
...
+        })
+
+        command.on('error', err => {
+            console.log(err);
+            reject()
+        })
+
+        command.on('close', (code) => {
+            var lines = data.split('\n');
+            var myRegexp = /Track ID (\d): (video|audio)/;
```
Comment (flokol120) [inline utils/downloader.js]: This RegExp is not working with any locale but en (en_XX). I.e.: Using a German locale the output looks like this: `Spur ID 0: video (MPEG-4p10/AVC/h.264)`. Changing the RegExp to `/ID (\d): (video|audio)/` seems to be working.

Comment (HardResetSolution) [reply utils/downloader.js]: strange... It worked under my German Windows, but your suggestion is more flexible either way.

**Comment by flokol120** [comment]
@HardResetSolution thank you very much!

**Pull request merged**